shared/elements/gaia_grid/js/items/grid_item.js

```diff
@@ -514,7 +514,14 @@
 
         var nameEl = document.createElement('span');
         nameEl.className = 'title';
-        nameEl.textContent = this.name;
+
+        if (this.name && this.name.then) {
```

Comment:
This looks a bit weird with a getter to me. I wonder if we should have an asyncName() method or something that we could check here instead.

```js
if (this.asyncName) {
  this.asyncName().then(...);
} else {
```